commit 0c91c12e799b1f84969822c8744d5041f408ea1f (HEAD -> master, refs/changes/01/4501/3)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2018-11-30T00:46:17+00:00 (10 months ago)
    
    TRITON-580 urclient could include some more details about who's running the command
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Approved by: Robert Mustacchi <rm@joyent.com>

diff --git a/lib/mq.js b/lib/mq.js
index a082a76..b2d8776 100644
--- a/lib/mq.js
+++ b/lib/mq.js
@@ -17,6 +17,7 @@ var mod_backoff = require('backoff');
 var mod_jsprim = require('jsprim');
 var mod_uuid = require('libuuid');
 var mod_verror = require('verror');
+var mod_os = require('os');
 
 var lib_common = require('./common');
 
@@ -614,6 +615,8 @@ send_ping(server_uuid)
 	return (infl);
 };
 
+var networkIps;
+
 URConnection.prototype.send_command = function
 send_command(server_id, script, args, env, data)
 {
@@ -623,6 +626,24 @@ send_command(server_id, script, args, env, data)
 		return (null);
 	}
 
+	if (networkIps === undefined) {
+		var netIfs = mod_os.networkInterfaces();
+		networkIps = [];
+		mod_jsprim.forEachKey(netIfs, function (ifName, ifAddrs) {
+			ifAddrs.forEach(function (ifAddr) {
+				if (!ifAddr.internal) {
+					networkIps.push(ifAddr.address);
+				}
+			});
+		});
+	}
+	var origin = {
+		user: process.env.LOGNAME,
+		smf_fmri: process.env.SMF_FMRI,
+		hostname: mod_os.hostname(),
+		ips: networkIps
+	};
+
 	var infl = self.urc_inflights.register(data);
 
 	self.urc_exchange.publish('ur.execute.' + server_id + '.' +
@@ -632,7 +653,8 @@ send_command(server_id, script, args, env, data)
 		args: args.map(function (x) {
 			return (x.replace(/%%ID%%/g, infl.id()));
 		}),
-		env: env
+		env: env,
+		origin: origin
 	});
 
 	return (infl);
diff --git a/package.json b/package.json
index 4f78d6d..5c66115 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "urclient",
-  "version": "1.2.0",
+  "version": "1.2.1",
   "description": "Triton ur client",
   "author": "Joyent (joyent.com)",
   "dependencies": {
